compose/src/main/java/com/bottlerocketstudios/compose/map/GoogleMapsView.kt

```diff
@@ -49,101 +49,141 @@ fun GoogleMapsView(googleMapScreenState: GoogleMapScreenState, toolbarEnabled: B
             }
         }
 
-        Button(onClick = {
-            mapUiSettings = mapUiSettings.copy(
-                mapToolbarEnabled = !mapUiSettings.mapToolbarEnabled
+        AnimatedVisibility(dialogVisibility.value) {
+            ShowErrorDialog(
+                yelpError = googleMapScreenState.yelpError.value,
+                onDismiss = {
+                    googleMapScreenState.resetError()
+                    showRetryButton.value = true
+                }
             )
-        }) {
-            Text(text = stringResource(R.string.toggle_map_toolbar))
         }
 
         AnimatedVisibility(googleMapScreenState.businessList.value.isNotEmpty()) {
-            YelpBusinessList(businessList = googleMapScreenState.businessList.value)
+            YelpBusinessList(
+                businessList = googleMapScreenState.businessList.value, onClick = { yelpBusiness ->
+                    googleMapScreenState.setSelectedMarker(
+                        YelpMarker(
+                            latitude = yelpBusiness.coordinates.latitude,
+                            longitude = yelpBusiness.coordinates.longitude,
+                            businessName = yelpBusiness.businessName
+                        )
+                    )
+                },
+                selectedYelpMarker = googleMapScreenState.yelpMarkerSelected.value
+            )
         }
     }
 }
 
-@OptIn(ExperimentalFoundationApi::class)
 @Composable
-fun ColumnScope.YelpBusinessList(businessList: List<Business>) {
-    LazyColumn(
-        verticalArrangement = Arrangement.spacedBy(Dimens.grid_1_5),
-        modifier = Modifier
-            .padding(
-                start = Dimens.grid_1_5,
-                end = Dimens.grid_1_5,
-                top = Dimens.grid_1_5,
-                bottom = Dimens.grid_1_5
-            )
-            .fillMaxWidth()
-            .weight(1f)
-    ) {
-        items(
-            items = businessList,
-            itemContent = { item ->
-                YelpCardLayout(
-                    business = item,
-                    selectItem = {},
-                    modifier = Modifier.animateItemPlacement()
-                )
-            }
+fun ShowErrorDialog(yelpError: UserFacingError, onDismiss: () -> Unit) {
+    when (yelpError) {
+        is UserFacingError.ApiError -> CustomAlertDialog(
+            modifier = Modifier,
+            title = yelpError.title,
+            message = yelpError.description,
+            onDismiss = onDismiss
+        )
+        is UserFacingError.GeneralError -> CustomAlertDialog(
+            modifier = Modifier,
+            title = yelpError.title,
+            message = yelpError.description,
+            onDismiss = onDismiss
+        )
+        else -> {
+            onDismiss()
+        }
+    }
+}
+@Composable
+fun addMarkers(yelpMarkers: List<YelpMarker>, onclick: (Marker) -> Boolean, yelpMarkerSelected: YelpMarker) {
```

Comment:
@Josh-Gebbeken @BottleRocket-Colin Should we use UpperCamelCase consistently for all Composable functions? If so, we could refactor this to be `AddMarkers`.